client/asset/eth/eth.go

```diff
@@ -2778,25 +2671,13 @@ func (w *assetWallet) ContractLockTimeExpired(ctx context.Context, contract dex.
 	return expired, swap.LockTime, nil
 }
 
-func (eth *baseWallet) addPendingTx(assetID uint32, txHash common.Hash, nonce, out, in, fees uint64) {
-	// We don't track pending txs locally if we have access to txpool.
-	if _, is := eth.node.(txPoolFetcher); is {
-		return
-	}
-	eth.tipMtx.RLock()
-	tip := eth.currentTip.Number.Uint64()
-	eth.tipMtx.RUnlock()
-	eth.pendingTxMtx.Lock()
-	eth.pendingTxs[txHash] = &pendingTx{
-		assetID:   assetID,
-		out:       out,
-		in:        in,
-		maxFees:   fees,
-		nonce:     nonce,
-		stamp:     time.Now(),
-		lastCheck: tip,
+func (eth *baseWallet) tip() uint64 {
+	ethWallet := eth.wallet(eth.baseChainID)
+	if ethWallet == nil || !ethWallet.connected.Load() {
```

Comment:
We don't usually do these types of protections, because the wallet should never require a tip unless it is connected. Did this solve a problem for you?

Reply:
I've removed it. It was allowing the tests to pass, but now I'm just setting the `currentTip` in the tests.